### PR TITLE
feat(tekton): tekton pipeline run list page

### PR DESCRIPTION
**Pull request by invincibleJai**
Tracks: https://github.com/janus-idp/backstage-plugins/issues/182

**Screenshots:**

image

image

**Specs:**

image

### CODE REVIEW

**Comment by debsmita1** [comment]
The empty state should show "No PipelineRuns found"
Screenshot 2023-04-10 at 11 38 02 PM


I don't see the pointer cursor when hovered over the task status bar

File: plugins/tekton/src/utils/tekton-utils.ts [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+  if (sec >= 60) {
+    min = Math.floor(sec / 60);
+    sec %= 60;
+  }
+  if (min >= 60) {
+    hr = Math.floor(min / 60);
+    min %= 60;
+  }
+  if (hr > 0) {
+    duration += long ? `${hr} hour` : `${hr}h`;
```
Comment (debsmita1) [inline plugins/tekton/src/utils/tekton-utils.ts:169]: Use the `pluralize` util from @patternfly/react-core

```suggestion
    duration += long ? pluralize(hr, 'hour', 'hours') : `${hr}h`;
```

**Comment by debsmita1** [comment]
The pipeline run status shows the reason, you might want to use the `pipelineRunStatus` util to show the proper status here

Screenshot 2023-04-11 at 12 02 58 AM

**Comment by karthikjeeyar** [comment]
Nit: The Timestamp field's font size and color looks different from other fields. 

Screenshot 2023-04-11 at 2 55 17 PM

File: plugins/tekton/src/utils/tekton-utils.ts [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    Running: 0,
+    Succeeded: 0,
+    Failed: 0,
+    Cancelled: 0,
+    Skipped: skippedTaskLength,
+  };
+  if (!pipelinerun?.status?.taskRuns) {
+    return taskStatus;
+  }
+  const plrTasks = Object.keys(pipelinerun.status.taskRuns);
```
Comment (karthikjeeyar) [inline plugins/tekton/src/utils/tekton-utils.ts:92]: we shouldn't be relying on this deprecated field anymore `pipelinerun.status.taskRuns` instead use the taskruns retrieved as a custom resource.

File: plugins/tekton/src/utils/tekton-utils.ts [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+  return {
+    ...taskStatus,
+  };
+};
+
+export const getTaskStatus = (pipelinerun: PipelineRunKind) => {
+  const totalTasks = totalPipelineRunTasks(pipelinerun);
+  const plrTasks = pipelinerun?.status?.taskRuns
+    ? Object.keys(pipelinerun.status.taskRuns)
```
Comment (karthikjeeyar) [inline plugins/tekton/src/utils/tekton-utils.ts:117]: same as above comment.

**Comment by invincibleJai** [comment]
> Nit: The Timestamp field's font size and color looks different from other fields.
> 
> Screenshot 2023-04-11 at 2 55 17 PM

Have updated this 

image

**Comment by karthikjeeyar** [comment]
LGTM!

**Comment by karthikjeeyar** [comment]
LGTM!

**Comment by debsmita1** [comment]
Task status is not updating 
backstage
Screenshot 2023-04-13 at 12 46 18 PM

console
Screenshot 2023-04-13 at 12 46 02 PM


Also, the status column values have the status icon along with the text, and can we also show the PLR resource badge ?

**Comment by debsmita1** [comment]
verified it locally

https://user-images.githubusercontent.com/22490998/232119994-e1e2cc82-0a38-4128-829b-959951bd7a4d.mov

LGTM !

**Pull request merged**